apps/open-swe/evals/evaluator.ts

```diff
@@ -0,0 +1,320 @@
+
+        const issueCount = Array.isArray(ruffIssues) ? ruffIssues.length : 0;
+        testResults.ruffScore = issueCount === 0 ? 1 : 0; // Binary scoring: pass/fail
+
+        logger.info(`Ruff found ${issueCount} issues`, {
+          score: testResults.ruffScore,
+          issues: ruffIssues.slice(0, 3), // Log first 3 issues
+        });
+      } catch (parseError) {
+        testResults.ruffScore = 0;
+        logger.warn(
+          "Could not parse ruff JSON output. Setting Ruff score to 0.",
+          {
+            parseError,
+            output: ruffRes.result?.substring(0, 200) + "...",
+          },
+        );
+      }
+    }
+  } catch (error) {
+    logger.error("Failed to run ruff check", { error });
+    testResults.details.ruff.error = error;
+    testResults.ruffScore = 0;
+  }
+
+  // Run mypy on all Python files in the repository
+  try {
+    logger.info("Running mypy type check on all Python files...");
+    const mypyRes = await sandbox.process.executeCommand(
+      `${RUN_PYTHON_IN_VENV} -m mypy . --no-error-summary --show-error-codes --no-color-output`,
+      absoluteRepoDir,
+      undefined,
+      TIMEOUT_SEC * 3,
+    );
+
+    testResults.details.mypy.exitCode = mypyRes.exitCode;
+    testResults.details.mypy.output = mypyRes.result;
+
+    if (mypyRes.exitCode === 0) {
+      testResults.mypyScore = 1; // Perfect score if no issues
+      logger.info("MyPy analysis passed - no type issues found");
+    } else {
+      const errorLines = mypyRes.result
+        .split("\n")
+        .filter(
+          (line) => line.includes(": error:") || line.includes(": warning:"),
+        );
+
+      testResults.details.mypy.issues = errorLines;
+
+      const issueCount = errorLines.length;
+      testResults.mypyScore = issueCount === 0 ? 1 : 0; // Binary scoring: pass/fail
+
+      logger.info(`MyPy found ${issueCount} issues`, {
+        score: testResults.mypyScore,
+        issues: errorLines.slice(0, 3),
+      });
+    }
+  } catch (error) {
+    logger.error("Failed to run mypy", { error });
+    testResults.details.mypy.error = error;
+    testResults.mypyScore = 0;
+  }
+
+  return testResults;
+}
+
+/**
+ * Main evaluator function for OpenSWE code analysis
+ */
+export async function evaluator(inputs: {
+  openSWEInputs: OpenSWEInput;
+  output: {
+    branchName: string;
+    targetRepository: TargetRepository;
+  };
+}): Promise<SimpleEvaluationResult[]> {
+  const { openSWEInputs, output } = inputs;
+
+  const githubToken = process.env.GITHUB_PAT;
+  if (!githubToken) {
+    throw new Error("GITHUB_PAT environment variable is not set");
+  }
+
+  const daytonaInstance = new Daytona();
+  logger.info("Creating sandbox...", {
+    repo: openSWEInputs.repo,
+    originalBranch: openSWEInputs.branch,
+    solutionBranch: output.branchName,
+    user_input: openSWEInputs.user_input.substring(0, 100) + "...",
+  });
+
+  const sandbox = await daytonaInstance.create({
+    image: SNAPSHOT_NAME,
+  });
```

Comment:
use the shared sandbox params:
```suggestion
  const sandbox = await daytonaInstance.create(DEFAULT_SANDBOX_CREATE_PARAMS);
```